client/src/components/ListCard.js

```diff
@@ -42,12 +43,17 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const ListCard = (props) => {
+  const {list, demoListsArray} = props;
   const { listTitle, itemCount, img } = props.list;
   const classes = useStyles();
   const [listOpen, setListOpen] = useState(false);
   const changeListOpen = () => {
     setListOpen(!listOpen);
   };
+  const [addProductOpen, setAddProductOpen] = useState(false);
+  const changeAddProductOpen = () => {
+    setAddProductOpen(!addProductOpen);
```

Comment:
With state manipulation that is dependent on the previous state (i.e. toggling a boolean, like this case) it's best to use the following syntax:

`setAddProductOpen((previous) => !previous);`   (where "previous" can be named anything)

This ensures the previous state value is used, as react setState is asynchronous so it is not usually guaranteed that the previous value will be updated in time ([explained here](https://stackoverflow.com/a/36087156)).